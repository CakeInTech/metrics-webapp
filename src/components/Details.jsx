import React, { useEffect } from 'react';
import { ListGroup } from 'react-bootstrap';
import { useSelector, useDispatch } from 'react-redux';
import { useLocation } from 'react-router-dom';
import { fetchDetails } from '../redux/countries/details';
import '../sass/Details.scss';

const CountryDetails = () => {
  const { countryDetails } = useSelector((state) => state.detailslice);
  const dispatch = useDispatch();
  const { state } = useLocation();

  useEffect(() => {
    dispatch(fetchDetails(state));
  }, []);

  return (
    <div className="countryDetails">
      {countryDetails.map((country) => (
        <div className="cardDetails" key={country.id}>
          <div className="detailsTop">
            <h2>{country.name.common}</h2>
            <img className="detailFlag" src={country.flags.svg} alt="Flag" />
          </div>
          <ListGroup className="items">
            <ListGroup.Item className="detailItem">
              Capital City:
              {country.capital}
            </ListGroup.Item>
            <ListGroup.Item className="detailItem">
              Area:
              {` ${country.area}km`}
            </ListGroup.Item>
            <ListGroup.Item className="detailItem">
              Population:
              {` ${country.population}`}
            </ListGroup.Item>
            <ListGroup.Item className="detailItem">
              independent:
              {` ${country.independent}`}
            </ListGroup.Item>
            <ListGroup.Item className="detailItem">
              Landlocked:
              {` ${country.landlocked}`}
            </ListGroup.Item>
          </ListGroup>
        </div>
      ))}
    </div>
  );
};

export default CountryDetails;
