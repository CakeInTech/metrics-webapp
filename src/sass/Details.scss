$primary-color: #ffff;
$lineA: #4368af;
$lineB-color: #3e63a5;

.countryDetails {
  width: 100%;

  .cardDetails {
    display: flex;
    flex-direction: column;
    gap: 3rem;
    padding: 20px;

    .detailsTop {
      display: flex;
      flex-direction: column;
      text-align: center;
      color: $primary-color;
    }

    .detailItem {
      color: $primary-color;
      background-color: $lineA;
    }

    .detailItem:nth-child(odd) {
      background-color: $lineB-color;
    }
  }
}
