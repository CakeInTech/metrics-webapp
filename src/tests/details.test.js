import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Details from '../components/Details';

describe('Rendering the navbar', () => {
  test('if the navbar renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Details />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
