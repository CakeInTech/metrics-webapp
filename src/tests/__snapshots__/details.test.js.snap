// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering the navbar if the navbar renders correctly 1`] = `
<div
  className="countryDetails"
/>
`;
